filebeat/harvester/reader/log.go

```diff
@@ -37,11 +37,11 @@ type LogFileReaderConfig struct {
 	CloseRemoved  bool
 }
 
-func NewLogFileReader(
+func NewLogFile(
 	fs source.FileSource,
-	config LogFileReaderConfig,
+	config LogFileConfig,
 	done chan struct{},
-) (*logFileReader, error) {
+) (*logFile, error) {
```

Comment:
exported `NewX` should not return unexported type. Can still be assigned using type inference and even converted to interface, but not be used by name.